st2exporter/st2exporter/worker.py

```
@@ -13,118 +13,65 @@
-from st2exporter.exporter.dumper import Dumper
+from st2exporter.consumers.execution import ExecutionsExporter
 
-__all__ = [
-    'ExecutionsExporter'
-]
-
-COMPLETION_STATUSES = [LIVEACTION_STATUS_SUCCEEDED, LIVEACTION_STATUS_FAILED,
-                       LIVEACTION_STATUS_CANCELED]
-LOG = logging.getLogger(__name__)
 
 EXPORTER_WORK_Q = execution.get_queue(
     'st2.exporter.work', routing_key=publishers.UPDATE_RK)
+LOG = logging.getLogger(__name__)
+
 
+class Exporter(object):
+    def __init__(self):
+        self._model_exporters = []
 
-class ExecutionsExporter(consumers.MessageHandler):
-    message_type = ActionExecutionDB
+        self._executions_exporter = None
+        with Connection(transport_utils.get_messaging_urls()) as conn:
+            self._executions_exporter = ExecutionsExporter(conn, [EXPORTER_WORK_Q])
 
-    def __init__(self, connection, queues):
-        super(ExecutionsExporter, self).__init__(connection, queues)
-        self.pending_executions = Queue.Queue()
-        self._dumper = Dumper(queue=self.pending_executions,
-                              export_dir=cfg.CONF.exporter.dump_dir)
-        self._consumer_thread = None
+        self._model_exporters.append(self._executions_exporter)
+        LOG.info('Available model exporters %s.', self._model_exporters)
+        self._model_exporters_pool = eventlet.GreenPool(len(self._model_exporters))
+        self._stopped = False
 
     def start(self, wait=False):
-        LOG.info('Bootstrapping executions from db...')
-        try:
-            self._bootstrap()
-        except:
-            LOG.exception('Unable to bootstrap executions from db. Aborting.')
-            raise
-        self._consumer_thread = eventlet.spawn(super(ExecutionsExporter, self).start, wait=True)
-        self._dumper.start()
+        LOG.info('Starting model exporters...')
+        for model_exporter in self._model_exporters:
+            while self._model_exporters_pool.free() <= 0 and not self._stopped:
```

Yeah, but as mentioned above, this would indicate a programmer error, right?

If so, we should fail fast and throw an exception.

Anyway.